tests/preliminary/test_petrification.py

```diff
@@ -0,0 +1,188 @@
+            "implementation_type": "AragonApp",
+        },
+        "VotingRepo": {
+            "contract": interface.AppProxyUpgradeable(contracts.voting_app_repo.address),
+            "proxy_type": "AppProxyUpgradeable",
+            "implementation_type": "AragonApp",
+        },
+        "OracleRepo": {
+            "contract": interface.AppProxyUpgradeable(contracts.oracle_app_repo.address),
+            "proxy_type": "AppProxyUpgradeable",
+            "implementation_type": "AragonApp",
+        },
+        # contracts with dummy impl to be upgraded with template
+        "Withdrawal_vault": {
+            "contract": interface.WithdrawalVaultManager(contracts.withdrawal_vault.address),
+            "proxy_type": "WithdrawalsManagerProxy",
+            "implementation_type": "Dummy",
+        },
+        "LidoLocator": {
+            "contract": interface.OssifiableProxy(contracts.lido_locator.address),
+            "proxy_type": "OssifiableProxy",
+            "implementation_type": "Dummy",
+        },
+        "StakingRouter": {
+            "proxy_type": "OssifiableProxy",
+            "implementation_type": "Dummy",
+            "contract": interface.OssifiableProxy(contracts.staking_router.address),
+        },
+        "WithdrawalQueue": {
+            "contract": interface.OssifiableProxy(contracts.withdrawal_queue.address),
+            "proxy_type": "OssifiableProxy",
+            "implementation_type": "Dummy",
+        },
+        "AccountingOracle": {
+            "contract": interface.OssifiableProxy(contracts.accounting_oracle.address),
+            "proxy_type": "OssifiableProxy",
+            "implementation_type": "Dummy",
+        },
+        "ValidatorsExitBusOracle": {
+            "contract": interface.OssifiableProxy(contracts.validators_exit_bus_oracle.address),
+            "proxy_type": "OssifiableProxy",
+            "implementation_type": "Dummy",
+        },
+        # pure implementations
+        "WithdrawalQueueImplementation": {
+            "contract": interface.WithdrawalQueueERC721(lido_dao_withdrawal_queue_implementation),
+            "proxy_type": "Implementation",
+            "implementation_type": "Versioned",
+        },
+        "WithdrawalVaultImplementation": {
+            "contract": interface.WithdrawalVault(lido_dao_withdrawal_vault_implementation),
+            "proxy_type": "Implementation",
+            "implementation_type": "Versioned",
+        },
+        "StakingRouterImplementation": {
+            "contract": interface.StakingRouter(lido_dao_staking_router_implementation),
+            "proxy_type": "Implementation",
+            "implementation_type": "Versioned",
+        },
+        "AccountingOracleImplementation": {
+            "contract": interface.AccountingOracle(lido_dao_accounting_oracle_implementation),
+            "proxy_type": "Implementation",
+            "implementation_type": "Versioned",
+        },
+        "ValidatorsExitBusOracleImplementation": {
```

Comment:
I'd prefer more straighforward and stupid design in the tests, because it's more readable and we don't expect dozens of new items like this. I mean something like
- define 3 function like: `check_aragon_petrified_implementation(proxy_address), check_ossifiable_proxy_dummy_impl(proxy_address), ...`
- make like 30 one liner calls

Reply:
I am not sure if that would be more readable, perhaps it's a style thing. But I agree that both test and config lack readability. I refactored them to be straightforward and robust. So a quick glance at test func would allow you to check config for correctness.
Removed some boilerplate, separated conditions and ordered them to match config, added comments and whitespace. 
Please have a look if you can.